optima/parameters.py

```diff
@@ -531,7 +531,7 @@ def interp(self, tvec, smoothness=20):
             output = zeros((npops,len(tvec)))
             for pop,key in enumerate(keys): # Loop over each population, always returning an [npops x npts] array
                 output[pop,:] = self.m * smoothinterp(tvec, self.t[pop], self.y[pop], smoothness=smoothness) # Use interpolation
-        if npops==1: return output[0,:]
+        if npops==1 and self.by!='pship': return output[0,:]
```

Comment:
so i assume this is to fix it in cases where there's a single partnership, but it'll still fail for a single population, so maybe `if npops==1 and self.by!='pship'` should be replaced by `if self.by=='tot'`?